src/field_theory/minpoly/gcd_monoid.lean

```diff
@@ -121,4 +121,81 @@ end
 
 end gcd_domain
 
+section adjoin_root
+
+noncomputable theory
+
+variables {R S : Type*} [comm_ring R] [comm_ring S] [algebra R S] (x : S) (R)
+
+open algebra polynomial adjoin_root
+
+/-- The surjective algebra morphism `R[X]/(minpoly R x) → R[x]`.
+If `R` is a GCD domain and `x` is integral, this is an isomorphism,
+see `adjoin_root.minpoly.equiv_adjoin`. -/
+@[simps] def to_adjoin : adjoin_root (minpoly R x) →ₐ[R] adjoin R ({x} : set S) :=
+lift_hom _ ⟨x, self_mem_adjoin_singleton R x⟩
+  (by simp [← subalgebra.coe_eq_zero, aeval_subalgebra_coe])
+
+variables {R x}
+
+lemma to_adjoin_apply' (a : adjoin_root (minpoly R x)) : to_adjoin R x a =
+  lift_hom (minpoly R x) (⟨x, self_mem_adjoin_singleton R x⟩ : adjoin R ({x} : set S))
+  (by simp [← subalgebra.coe_eq_zero, aeval_subalgebra_coe]) a := rfl
+
+lemma to_adjoin.apply_X : to_adjoin R x (mk (minpoly R x) X) =
+  ⟨x, self_mem_adjoin_singleton R x⟩ :=
+by simp
+
+variables (R x)
+
+lemma to_adjoin.surjective : function.surjective (to_adjoin R x) :=
+begin
+  rw [← range_top_iff_surjective, _root_.eq_top_iff, ← adjoin_adjoin_coe_preimage],
+  refine adjoin_le _,
+  simp only [alg_hom.coe_range, set.mem_range],
+  rintro ⟨y₁, y₂⟩ h,
+  refine ⟨mk (minpoly R x) X, by simpa using h.symm⟩
+end
```

Comment:
This doesn't seem to require any GCD stuff. Should it be in this file?

Reply:
Good point! I've moved those back